### PR TITLE
Logo link becomes a permalink

### LINKED ISSUE #1429: permalink
the link in the bottom right of the API map should be a permalink to the map (X,Y,Zoom, background and overlay layers

### PR DESCRIPTION
**Pull request by pgiraud**
Fixes #1429.
Please review.

### CODE REVIEW

File: jsapi/externs/luxx.js [hunk 1]
```diff
@@ -285,3 +285,45 @@ luxx.FeaturesOptions.prototype.ids;
...
+
+/**
+ * @typedef {{
+ *   X: (number|undefined),
+ *   Y: (number|undefined),
+ *   zoom: (number|undefined),
+ *   layers: (Array<number>|undefined),
+ *   opacities: (Array<number>|undefined),
+ *   bgLayer: (string|undefined)
+ * }}
```
Comment (tonio) [inline jsapi/externs/luxx.js]: We just use `{Object}` for other objects to avoid type duplication, isn't it?

File: jsapi/src/statemanager.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    var zoom = view.getZoom();
+    var object = {};
+    object['X'] = Math.round(center[0]);
+    object['Y'] = Math.round(center[1]);
+    object['zoom'] = zoom;
+    this.updateState(object);
+  }
+
+  // make sure the state is updated the first time
+  onViewUpdate.call(this);
```
Comment (tonio) [inline jsapi/src/statemanager.js]: `onViewUpdate()`?

File: jsapi/src/statemanager.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  ol.events.listen(view, 'propertychange',
+      lux.debounce(onViewUpdate, 250).bind(this));
+
+  var layersListenerKeys = [];
+
+  function onLayersUpdate() {
+    // remove any listener on visibility change
+    layersListenerKeys.forEach(function(key) {
+      ol.events.unlistenByKey(key);
+    });
```
Comment (tonio) [inline jsapi/src/statemanager.js]: Shouldn’t `layersListenerKeys` be emptied?

File: jsapi/src/statemanager.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+          this);
+      }
+    }.bind(this));
+    object['layers'] = layers.join('-');
+    object['opacities'] = opacities.join('-');
+    this.updateState(object);
+  }
+
+  ol.events.listen(map.getLayers(), ol.CollectionEventType.ADD,
+    onLayersUpdate, this);
```
Comment (tonio) [inline jsapi/src/statemanager.js]: Should be called on `REMOVE`, too.

File: jsapi/src/statemanager.js (hunk 2 above)
Comment (pgiraud) [inline jsapi/src/statemanager.js]: I need to specify the scope.

File: jsapi/src/statemanager.js (hunk 3 above)
Comment (pgiraud) [inline jsapi/src/statemanager.js]: Indeed.

File: jsapi/src/statemanager.js (hunk 4 above)
Comment (pgiraud) [inline jsapi/src/statemanager.js]: I didn't expect layer to be removed. However, you're right, this should be handled.

**Pull request merged**